src/lib/features/metrics/client-metrics/metrics-service-v2.ts

```diff
@@ -178,12 +179,75 @@ export default class ClientMetricsServiceV2 {
         return toggleNames;
     }
 
+    private async siftMetrics(
```

Comment:
This method essentially extracts what we were already doing for filtering/validating in our `registerClientMetrics` method, makes it work for `IClientMetricsEnv[]` instead and optimizes it.